src/contexts/DataContext/DataContext.jsx

```diff
@@ -87,28 +110,45 @@ const DataProvider = ({ children }) => {
     ? [...allium, ...cruciferous, ...marrow, ...fruits, ...leafy, ...root]
     : priceFiltered;
 
-  const ratingfiltered=data.rating===0?categoryfiltered:categoryfiltered.filter(item=>item.rating>data.rating)
+  const season = data.alltime || data.winter || data.summer;
+
+  const alltime = categoryfiltered.filter((item) =>
+    item.season === "AllTime" && data.alltime ? true : false
+  );
+  const summer = categoryfiltered.filter((item) =>
+    item.season === "Summer" && data.summer ? true : false
+  );
+  const winter = categoryfiltered.filter((item) =>
+    item.season === "Winter" && data.winter ? true : false
+  );
+
+  const seasonFiltered = season
+    ? [...alltime, ...summer, ...winter]
+    : categoryfiltered;
+
+  const ratingfiltered =
+    data.rating === 0
+      ? seasonFiltered
+      : seasonFiltered.filter((item) => item.rating > data.rating);
 
   function getSorted(product, sortBy) {
     const output =
       sortBy === null
```

Comment:
don't use ternary operator for multi-line statements as it reduces the code readability